### PR TITLE
Change HasSBOM to Package or Artifact.

### PR DESCRIPTION
**Pull request by jeffmendoza**
Fixes: #786

### CODE REVIEW

**Review by pxp928 — APPROVED** [review]
LGTM! Just need to change the examples to match

**Review by lumjjb — COMMENTED** [review]
LGTM but will defer to @mihaimaruseac on another gql approval

File: docs/GraphQL.md [hunk 1]
```diff
@@ -601,7 +601,7 @@ packages and sources:
 type HasSBOM {
   id: ID!
   "SBOM subject"
-  subject: PackageOrSource!
+  subject: PackageOrArtifact!
```
Comment (mihaimaruseac) [inline docs/GraphQL.md:604]: You also need to change on line 598 in text, but the rest looks good

**Review by mihaimaruseac — APPROVED** [review]

File: pkg/assembler/backends/inmem/hasSBOM.go [hunk 2]
```diff
@@ -296,12 +292,9 @@ func (c *demoClient) addHasSBOMIfMatch(out []*model.HasSbom,
...
 	[]*model.HasSbom, error) {
 
-	algorithm := strings.ToLower(nilToEmpty(filter.Algorithm))
-	digest := strings.ToLower(nilToEmpty(filter.Digest))
-
 	if noMatch(filter.URI, link.uri) ||
-		noMatch(ptrfrom.String(algorithm), link.algorithm) ||
-		noMatch(ptrfrom.String(digest), link.digest) ||
+		noMatch(toLower(filter.Algorithm), link.algorithm) ||
+		noMatch(toLower(filter.Digest), link.digest) ||
```
Comment (jeffmendoza) [inline pkg/assembler/backends/inmem/hasSBOM.go:297]: These pointers were always non-nil. I'll get those unit tests in next ! =)

File: docs/GraphQL.md (hunk 1 above)
Comment (jeffmendoza) [reply docs/GraphQL.md:604]: done.

**Pull request merged**